src/tests/test_testing.py

```diff
@@ -1,7 +1,52 @@
 import pathlib
+import pytest
 
 
 def test_web_server_secondary(selenium, web_server_secondary):
     host, port, logs = web_server_secondary
     assert pathlib.Path(logs).exists()
     assert selenium.server_port != port
+
+
+def assert_C_test(result):
+    print("result:", repr(result))
+    if result:
+        raise AssertionError(result)
+
+
+def test_C_test_entrypoints(selenium):
+    assert selenium.run_js("return pyodide.Tests.test_entrypoints() === 'It works!';")
+
+
+def test_C_tests_succeed1(selenium):
+    assert_C_test(
+        selenium.run_js("return pyodide.Tests.c_tests_expect_success_success();")
+    )
+
+
+def test_C_tests_succeed2(selenium):
+    assert_C_test(selenium.run_js("return pyodide.Tests.c_tests_expect_fail_fail();"))
```

Comment:
The somewhat non ideal thing about it is that one has to first write a test in `src/type_conversion/testing.c`, and then some boilerplate code in this file. It would be better if pytest would just look at `pyodide.Tests` methods, and run those that have the `test` substring.Excluding tests that are expected to fail -- those are only to test the framework, right?

For instance, something like,
```py
@pytest.mark.parametrize(
    'name',
    ['c_tests_expect_success_success',
     'c_tests_expect_fail_fail',
     # add future tests here
     ]
)
def test_core_c_func(name, selenium):
     assert_C_test(selenium.run_js(f"return pyodide.Tests.{name}();"))
```
would be a start. There might be some more automated way of listing  all methods in `pyodide.Tests` maybe by using [parametrized fixtures](https://docs.pytest.org/en/stable/fixture.html#fixture-parametrize) but I don't see how to easily do that while also using the selenium fixture to generate the list of tests.

Reply:
I agree, I just don't understand pytest well enough to know how to do this. I can set it up as you show above. I was thinking we could add a make job to grep for `DEFINE_TEST_EXPECT_FAIL(` and `DEFINE_TEST(` and stick them into an autogenerated test file.

Reply:
yes, you can probably also just run regexp in Python the .c file and then auto-generate tests with the [`pytest_generate_tests` hook](https://docs.pytest.org/en/stable/parametrize.html#basic-pytest-generate-tests-example) as done [here](https://github.com/iodide-project/pyodide/blob/115507754122d77c81d987f2f2c1210ab205d98e/src/tests/test_core_python.py#L50)

Reply:
Thanks!

Reply:
So I guess what I will do is manually wrap the test tests, because some of the tests have to fail in order to cover the failure cases in the testing code. In the autogen we can just skip grepping `testing.c`